returnn/tf/layers/basic.py

```diff
@@ -6418,9 +6432,12 @@ class SubnetworkLayer(LayerBase):
   recurrent = True  # we don't know. depends on the subnetwork.
 
   # noinspection PyShadowingNames
-  def __init__(self, subnetwork, concat_sources=True, load_on_init=None,
+  def __init__(self, subnetwork,
+               _subnet, _output,
```

Comment:
These "internal" parameters are used to pass variables from the class method transform_config_dict to the instance? Seems a bit hacky.

Reply:
But that is the whole purpose of `transform_config_dict`.
I just keep the convention that internal arguments (which are not set by the user in the config) are prefixed with `_`.